File: src/bot/commands/nowplaying.ts
import type { Message } from 'discord.js-selfbot-v13';
import streamManager from '../../stream/manager.js';
import { formatDuration } from '../../plex/library.js';

export async function nowPlayingCommand(message: Message, _args: string[]): Promise<void> {
  if (!message.guild) {
    await message.edit('❌ This command can only be used in a server');
    return;
  }

  const guildId = message.guild.id;
  const state = streamManager.getState(guildId);

  if (!state) {
    await message.edit('❌ Nothing is currently playing');
    return;
  }

  const { current, total, percentage } = streamManager.getProgress(guildId);

  let title = state.mediaItem.title;
  if (state.mediaItem.type === 'episode' && state.mediaItem.grandparentTitle) {
    const season = state.mediaItem.parentIndex
      ? `S${String(state.mediaItem.parentIndex).padStart(2, '0')}`
      : '';
    const episode = state.mediaItem.index
      ? `E${String(state.mediaItem.index).padStart(2, '0')}`
      : '';
    title = `${state.mediaItem.grandparentTitle} ${season}${episode} - ${state.mediaItem.title}`;
  }

  const status = state.isPaused ? '⏸️ Paused' : '▶️ Playing';
  const progress = `${formatDuration(current)} / ${formatDuration(total)}`;
  const progressBar = createProgressBar(percentage);

  const info = [
    `🎬 **${title}**`,
    ``,
    `${status}`,
    `${progressBar}`,
    `⏱️ ${progress} (${percentage.toFixed(1)}%)`,
  ];

  if (state.mediaItem.year) {
    info.push(`📅 ${state.mediaItem.year}`);
  }

  await message.edit(info.join('\n'));
}

function createProgressBar(percentage: number): string {
  const total = 20;
  const filled = Math.round((percentage / 100) * total);
  const empty = total - filled;

  return `[${'▓'.repeat(filled)}${'░'.repeat(empty)}]`;
}

File: selfbot/commands/nowplaying.py
"""Now playing command via IPC."""

from __future__ import annotations

from typing import TYPE_CHECKING
import discord

from ..ipc import ipc
from ..helpers import format_duration, create_progress_bar

if TYPE_CHECKING:
    from ..client import SchroStreamBot


async def now_playing_command(bot: SchroStreamBot, message: discord.Message, args: list[str]) -> None:
    if not message.guild:
        await message.channel.send("❌ This command can only be used in a server")
        return

    result = await ipc.get_now_playing(str(message.guild.id))

    if not result.get("success"):
        await message.channel.send("❌ Nothing is currently playing")
        return

    data = result
    title = data.get("title", "Unknown")
    is_paused = data.get("isPaused", False)
    current = data.get("currentTime", 0)
    total = data.get("duration", 0)
    percentage = data.get("percentage", 0.0)

    status = "⏸️ Paused" if is_paused else "📺 Streaming (Go Live)"
    progress = f"{format_duration(current)} / {format_duration(total)}"
    bar = create_progress_bar(percentage)

    lines = [
        f"🎬 **{title}**",
        "",
        status,
        bar,
        f"⏱️ {progress} ({percentage:.1f}%)",
    ]

    year = data.get("year")
    if year:
        lines.append(f"📅 {year}")

    await message.channel.send("\n".join(lines))
